### DISCUSSION TITLE
"Skipping schema validation because no schema file has been set" error raised in Validator

### DISCUSSION THREAD

**Discussion by hrw-jla** [INSPIRE-MIF/helpdesk#98] [Q&A] [answered]
Morning, I am attempting to test a newly created GML file for INSPIRE compliance via the Validator and am receiving a fairly unhelpful error.
The file I am validating is a GML for the Annex III, Natural Risk Zones theme, and it passes an offline validation against the NZ Core XSD. The test log contains the message: _Skipping schema validation because no schema file has been set in the test suite. Data are only checked for well-formedness._ - could this be the reason for the error?
I would appreciate any help on this as we are the data providers for one of the EU member states' compliance with the EU Floods Directive. I attach a sample file that passes validation against the NZ Core XSD but just receives a generic error on the web validator.
[NZ_Sample.zip](https://github.com/INSPIRE-MIF/helpdesk/files/8777434/NZ_Sample.zip)

**Comment by sMorrone** [comment]
Dear @hrw-jla, 
this helpdesk is meant for discussions and announcements on general INSPIRE topics, while problems or questions relevant to specific INSPIRE components should be posted in the dedicated helpdesks (in this case, the [Validator helpdesk](https://github.com/INSPIRE-MIF/helpdesk-validator)). 
That said, I had a quick look at your sample file, and it is indeed missing declaration of schema location element, and that is why the Validator complains "_Skipping schema validation because no schema file has been set_ ". To solve the issue, you should add the following
`xsi:schemaLocation="http://inspire.ec.europa.eu/schemas/nz-core/4.0 https://inspire.ec.europa.eu/schemas/nz-core/4.0/NaturalRiskZonesCore.xsd" `
to the <gml:FeatureCollection> tag (below the declarations of the namespaces and the gml:id). 
Hope this helps
Stefania

**Comment by sMorrone** [comment]
I am changing the title of this discussion to "Skipping schema validation because no schema file has been set" error in the Validator"  in order to make it immediately clear which is the addressed issue.

**Reply by hrw-jla** [reply]
Thanks @sMorrone, but this hasn't solved the error - could you possibly provide me with a corrected version of the file which does validate? Thanks, James

**Comment by sMorrone** [comment, marked as answer]
Dear @hrw-jla
I modified the schema location element in your file and validated it (leaving only one feature). The  "Skipping schema validation because no schema file has been set" error does not appear any more. There are other validation errors, but these are not related to missing schema location. Find uploaded sample file which has no error related to missing schema.

 
[floodHazardRiskAssessment_INSPIRE_GML_BG1_APSFR_DB_100_Sample.zip](https://github.com/INSPIRE-MIF/helpdesk/files/8778389/floodHazardRiskAssessment_INSPIRE_GML_BG1_APSFR_DB_100_Sample.zip)
